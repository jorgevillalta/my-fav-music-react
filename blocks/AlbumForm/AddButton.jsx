import { Fab } from '@material-ui/core';
import AddIcon from '@material-ui/icons/Add';
import * as React from 'react';
import styled from 'styled-components';

const AddButton = styled(({ ...props }) => (
  <Fab color="primary" aria-label="Add" type="submit" {...props}>
    <AddIcon />
  </Fab>
))`
  && {
    flex: 0 0 auto;
    max-width: 40px;
    max-height: 40px;
  }
`;

export default AddButton;
