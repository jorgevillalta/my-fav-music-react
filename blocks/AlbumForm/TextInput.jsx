import { FormControl, TextField } from '@material-ui/core';
import * as React from 'react';
import styled from 'styled-components';

const TextInput = styled(({ ...props }) => (
  <FormControl>
    <TextField margin="normal" {...props} />
  </FormControl>
))``;

export default TextInput;
