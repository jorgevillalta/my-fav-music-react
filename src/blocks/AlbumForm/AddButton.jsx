import { Fab } from '@material-ui/core';
import AddIcon from '@material-ui/icons/Add';
import * as React from 'react';
import styled from 'styled-components';

const AddButton = styled(({ size, ...props }) => (
  <Fab
    color="primary"
    aria-label="Add"
    type="submit"
    size={size ? size : 'small'}
    {...props}>
    <AddIcon />
  </Fab>
))``;

export default AddButton;
