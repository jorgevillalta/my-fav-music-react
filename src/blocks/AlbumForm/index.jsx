import styled from 'styled-components';

import Content from './Content';
import Control from './Control';

const AlbumForm = styled.form``;

AlbumForm.Content = Content;
AlbumForm.Control = Control;

export default AlbumForm;
