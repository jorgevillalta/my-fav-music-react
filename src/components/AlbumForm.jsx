import * as React from 'react';
import { observer, inject } from 'mobx-react';
import { FormControl, TextField, Fab } from '@material-ui/core';
import AddIcon from '@material-ui/icons/Add';
import styled from 'styled-components';

import AlbumModel from '../models/AlbumModel';

const StyledDiv = styled.div`
  && {
    display: flex;
    align-content: stretch;
    align-items: center;

    /* childs margin */
    & > * {
      margin: 12px;
    }
  }
`;

@inject('albumStore')
@observer
class TodoForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      artist: '',
      name: ''
    };
  }

  handleInputChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleFormSubmit = e => {
    e.preventDefault();
    this.props.albumStore.add(
      new AlbumModel(this.state.artist, this.state.name)
    );
  };

  render() {
    return (
      <form onSubmit={this.handleFormSubmit}>
        <StyledDiv>
          <FormControl>
            <TextField
              label="Artist"
              name="artist"
              value={this.state.artist}
              onChange={this.handleInputChange}
              margin="normal"
            />
          </FormControl>
          <FormControl>
            <TextField
              label="Album"
              name="name"
              value={this.state.name}
              onChange={this.handleInputChange}
              margin="normal"
            />
          </FormControl>
          <Fab color="primary" size="small" aria-label="Add" type="submit">
            <AddIcon />
          </Fab>
        </StyledDiv>
      </form>
    );
  }
}

export default TodoForm;
