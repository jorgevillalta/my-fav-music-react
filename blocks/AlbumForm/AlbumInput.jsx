import * as React from 'react';
import styled from 'styled-components';

import TextInput from './TextInput';

const AlbumInput = styled(({ ...props }) => (
  <TextInput label={'Album ðŸ’¿'} name={'name'} {...props} />
))`
  && {
    flex: 1 1 auto;
  }
`;

export default AlbumInput;
