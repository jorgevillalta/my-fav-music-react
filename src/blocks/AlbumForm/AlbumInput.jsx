import * as React from 'react';
import styled from 'styled-components';

import TextInput from './TextInput';

const AlbumInput = styled(({ ...props }) => (
  <TextInput label={'Album ðŸ’¿'} name={'name'} {...props} />
))``;

export default AlbumInput;
