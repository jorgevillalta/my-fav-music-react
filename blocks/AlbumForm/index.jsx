import styled from 'styled-components';

import AddButton from './AddButton';
import AlbumInput from './AlbumInput';
import ArtistInput from './ArtistInput';
import Content from './Content';
import TextInput from './TextInput';

const AlbumForm = styled.form``;

AlbumForm.Content = Content;
AlbumForm.TextInput = TextInput;
AlbumForm.ArtistInput = ArtistInput;
AlbumForm.AlbumInput = AlbumInput;
AlbumForm.AddButton = AddButton;

export default AlbumForm;
