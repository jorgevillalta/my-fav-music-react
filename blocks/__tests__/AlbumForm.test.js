import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import * as React from 'react';

import AlbumForm from '../AlbumForm';

import 'jest-styled-components';

Enzyme.configure({ adapter: new Adapter() });

describe('AlbumForm Block', () => {
  let wrapper;
  const props = { name: 'input', size: 'small' };

  test('Renders ok', () => {
    wrapper = shallow(<AlbumForm />);
    expect(wrapper.exists()).toBeTruthy();
  });

  test('Match snapshot', () => {
    wrapper = mount(<AlbumForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('TextInput render props', () => {
    wrapper = mount(<AlbumForm.TextInput {...props} />);
    expect(wrapper.prop('name')).toEqual(props.name);
  });

  test('ArtistInput render props', () => {
    wrapper = mount(<AlbumForm.ArtistInput {...props} />);
    expect(wrapper.prop('name')).toEqual(props.name);
  });

  test('AlbumInput render props', () => {
    wrapper = mount(<AlbumForm.AlbumInput {...props} />);
    expect(wrapper.prop('name')).toEqual(props.name);
  });

  test('AddButton render props', () => {
    wrapper = mount(<AlbumForm.AddButton {...props} />);
    expect(wrapper.prop('name')).toEqual(props.name);
  });
});
