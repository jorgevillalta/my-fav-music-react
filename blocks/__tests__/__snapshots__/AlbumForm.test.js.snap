// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumForm Block Match snapshot 1`] = `
<styled.form>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "AddButton": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  && {
    flex: 0 0 auto;
    max-width: 40px;
    max-height: 40px;
  }
",
            ],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "AlbumInput": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  && {
    flex: 1 1 auto;
  }
",
            ],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "ArtistInput": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "
  && {
    flex: 1 1 auto;
  }
",
            ],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "Content": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
  && {
    display: flex;
    flex-direction: row;
    align-content: stretch;
    align-items: center;

    /* childs margin */
    & > * {
      margin: 12px;
    }
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "TextInput": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": true,
          "lastClassName": "jvzENE",
          "rules": Array [],
        },
        "displayName": "styled.form",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "form",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <form
      className=""
    />
  </StyledComponent>
</styled.form>
`;
